lib/elixir/lib/calendar/time.ex

```diff
@@ -679,14 +686,29 @@ defmodule Time do
 
       iex> Time.diff(~T[00:29:12], ~T[00:29:10], :microsecond)
       2_000_000
+
       iex> Time.diff(~T[00:29:10], ~T[00:29:12], :microsecond)
       -2_000_000
 
+      iex> Time.diff(~T[12:30:12], ~T[11:00:00], :minute)
+      90
+
+      iex> Time.diff(~T[12:30:12], ~T[11:00:00], :hour)
+      1
+
   """
   @doc since: "1.5.0"
-  @spec diff(Calendar.time(), Calendar.time(), System.time_unit()) :: integer
+  @spec diff(Calendar.time(), Calendar.time(), time_unit()) :: integer
   def diff(time1, time2, unit \\ :second)
 
+  def diff(%{calendar: Calendar.ISO} = time1, %{calendar: Calendar.ISO} = time2, :hour) do
+    time1.hour - time2.hour
+  end
+
+  def diff(%{calendar: Calendar.ISO} = time1, %{calendar: Calendar.ISO} = time2, :minute) do
+    (time1.hour - time2.hour) * 60 + (time1.minute - time2.minute)
```

Comment:
This will return the wrong result for this:

    Time.diff(~T[00:01:20], ~T[00:00:40])

It should return zero. I am afraid you will have to add all seconds and microseconds together, as in the clause below, and only then extracts minutes and hours.

Reply:
Excellent point. It felt like something was missing here.

Reply:
I've modified it to use the base `diff` implementation and `div/2` the result 👍